### PR TITLE
Log Account Linking

### PR DESCRIPTION
**Pull request by mborof**
Hey Dima.  These are the changes I discussed with you in the elevator.  If you could help certify the timing of each event I logged / make sure its appropriate.

I was in particular concerned about EVENT_LINKED_ACCOUNTS as I wasn't exactly sure the earliest moment I could log that.

### CODE REVIEW

**Comment by mborof** [comment]
Also note that tests will still fail here because I haven't released the API with the new enum values yet and haven't updated the tests.  I'm just hoping to validate the basic code flow.

File: src/runtime/link-accounts-flow.js [hunk 1]
```diff
@@ -186,6 +206,9 @@ export class LinkCompleteFlow {
    * @private
    */
   complete_(response) {
+    this.deps_
```
Comment (dvoytenko) [inline src/runtime/link-accounts-flow.js]: Why DISMISS here?

File: src/runtime/link-accounts-flow.js [hunk 2]
```diff
@@ -97,11 +102,17 @@ export class LinkCompleteFlow {
       );
       return promise.then(
         response => {
+          deps
+            .eventManager()
+            .logSwgEvent(AnalyticsEvent.ACTION_LINK_CONTINUE, true);
```
Comment (dvoytenko) [inline src/runtime/link-accounts-flow.js]: Not sure CONTINUE is the right word here. But may be it's consistent with other names? But the flow doesn't really continue here - it just completes. As in "the action is done" by this time. There are just UX considerations at this point, I believe.

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: Ideally I would log this when the user clicks the continue button.  Is there a better location to do that?

I am open to a better naming convention.  The intention is that the user clicked the button to proceed to the next step (of account linking)

File: src/runtime/link-accounts-flow.js (hunk 1 above)
Comment (mborof) [reply src/runtime/link-accounts-flow.js]: The user clicked a button "dismissing" / "closing" the screen thanking them for updating google with their account information

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: I'm open to a better name if you have an idea

Comment (dvoytenko) [reply src/runtime/link-accounts-flow.js]: DISMISS sounds like "cancel" to me. So maybe just CLOSE or COMPLETE?

File: src/runtime/link-accounts-flow.js (hunk 2 above)
Comment (dvoytenko) [reply src/runtime/link-accounts-flow.js]: That's the thing: user actually is not proceeding to the next step of account linking here IIUC. The user is all done - they are just acknowledging this fact. So maybe ACTION_LINK_ACK?

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: I definitely would like to log that the user clicked continue when they click continue.  Where can I log that?

Comment (dvoytenko) [reply src/runtime/link-accounts-flow.js]: I think logging it is good. Just meaning of this event is "acknowledgement" of an action, not "continuation" of an action.

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: If the current meaning of the event is acknowledgement then it is being logged in the wrong place.  We need something to be logged when the user clicks the button to continue and initiate the actual linking flow.  Please clarify where I would log this event in code.

Comment (dvoytenko) [reply src/runtime/link-accounts-flow.js]: Ok I might be misunderstanding this to some extent. Let's look at the actual flows.

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: Note that flow.start here will already log EVENT_GOOGLE_UPDATED and IMPRESSION_GOOGLE_UPDATED so we don't need a second thing logging that.  We just need something that logs that the user clicked the continue with linking button

Comment (mborof) [reply src/runtime/link-accounts-flow.js]: We discussed and we will move this logging to the server in a separate pull

**Comment by mborof** [comment]
The name of the dismiss event was changed to close.

We have no method of logging the user clicking next or canceling the workflow.  This will be fixed in a separate PR.

**Pull request merged**